services/web/server/src/simcore_service_webserver/version_control/db.py

```diff
@@ -137,11 +137,11 @@ async def _fetch_workcopy_project_id(
         )
         for tag in found:
             if workcopy_project_id := parse_workcopy_project_tag_name(tag.name):
-                return str(workcopy_project_id)
+                return ProjectIDStr(workcopy_project_id)
 
         repo = await self.ReposOrm(conn).set_filter(id=repo_id).fetch("project_uuid")
         assert repo  # nosec
-        return repo.project_uuid
+        return ProjectIDStr(repo.project_uuid)
```

Comment:
NOTE that using the constractor or a constrainted str type simply calls the constructor of `str` but does NOT run any validation. For that you need to do `parse_obj_as(ProjectIDStr, repo.project_uuid)`

Reply:
normally this comes with sqlalchemy[mypy]